candi/openapi/node_group.yaml

```diff
@@ -1905,6 +1905,41 @@ spec:
                         How many rotated log files to store before deleting them.
 
                         > **WARNING!** This parameter does nothing if CRI type is `Docker`.
+                    resourceReservationMode:
+                      type: string
+                      default: "Auto"
+                      enum: ["Off", "Auto", "Static"]
+                      description: |
+                        Specify whether to:
+
+                        * Disable resource reservation
+                        * Reserve resources based on the Node capacity.
+                        * Provide your own resource reservation values via the `staticResourceReservation` field.
+
+                        More info: https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#system-reserved
+
+                        Note that currently we do not use a dedicated cgroup for resource reservation (`-system-reserved-cgroup` is not used).
+                    staticResourceReservation:
+                      type: object
+                      properties:
+                        cpu:
+                          x-kubernetes-int-or-string: true
+                          anyOf:
+                            - type: integer
+                            - type: string
+                          pattern: '\d+[m]'
+                        memory:
+                          x-kubernetes-int-or-string: true
+                          anyOf:
+                            - type: integer
+                            - type: string
+                          pattern: '\d+[Ei|Pi|Ti|Gi|Mi|Ki|E|P|T|G|M|k|m]'
+                        ephemeralStorage:
+                          x-kubernetes-int-or-string: true
+                          anyOf:
+                            - type: integer
+                            - type: string
+                          pattern: '\d+[Ei|Pi|Ti|Gi|Mi|Ki|E|P|T|G|M|k|m]'
```

Comment:
it's very strange nodeGroup config for me:
```
nodegroup:
  resourceReservationMode: Static
  staticResourceReservation:
    ...
```
May be it can be something like 
```
nodegroup:
  reservation:
    mode: Static
    static:
       ...
```
 with oneOF checking (if mode: Static than static key should be present)

Reply:
Fixed.